### PR TITLE
[BUG_FIX][TOPI] Allow topi resize to accept more options

### PR DESCRIPTION
**Pull request by electriclilies**
In PR #6298, we restricted the options supported by topi resize. This was incorrect and was causing failures in the onnx importer tests for resize. This PR removes the check added in #6289, and adds more tests to topi and onnx. 

Finally, the topi testing suite can't produce reference values for resize with the options nearest_neighbor and align_corners, so I skip that combo in the topi and relay op tests. However, these options work in the onnx tests, so it's not a huge concern. 

@mbrookhart Please take a look!

### CODE REVIEW

File: tests/python/frontend/onnx/test_forward.py [hunk 1]
```diff
@@ -3356,15 +3356,19 @@ def verify(ishape, oshape, scales, mode, coord_trans):
 
     # upsampling
     verify([1, 16, 32, 32], [1, 16, 64, 64], [], "nearest", "asymmetric")
+    verify([1, 16, 32, 32], [1, 16, 64, 64], [], "linear", "asymmetric")
+    verify([1, 16, 32, 32], [1, 16, 64, 64], [], "nearest", "align_corners")
     verify([1, 16, 32, 32], [1, 16, 64, 64], [], "linear", "align_corners")
+    verify([1, 16, 32, 32], [1, 16, 64, 64], [], "nearest", "half_pixel")
     verify([1, 16, 32, 32], [1, 16, 64, 64], [], "linear", "half_pixel")
-    # downsampling
```
Comment (mbrookhart) [inline tests/python/frontend/onnx/test_forward.py:3361]: Why remove down-sampling tests?

**Review by mbrookhart — COMMENTED** [review]
Minor nitpick on one of the tests, but otherwise looks great! Thanks.

File: tests/python/frontend/onnx/test_forward.py (hunk 1 above)
Comment (electriclilies) [reply tests/python/frontend/onnx/test_forward.py:3361]: Whoops, I thought they were the same as the upsampling tests (I can read).. I'll add them back

**Review by mbrookhart — APPROVED** [review]

**Comment by mbrookhart** [comment]
Thanks @electriclilies

**Comment by electriclilies** [comment]
Thanks @mbrookhart :)

**Pull request merged**